Folder names with spaces aren't processed correctly on Windows 10 Pro

**V version:** V 0.1.18 
**OS:** Windows 10 Pro 1809 Build 17763.678

**What did you do?**
`e:\Program Files\vlang\v>v run examples\news_fetcher.v`

**What did you expect to see?**
Some news...
 
**What did you see instead?**
```v Files\vlang\v\thirdparty\cJSON\cJSON.o not found, building it...
ls() couldnt open dir "Files\vlang\v\thirdparty\cJSON" (does not exist).
Der Befehl "gcc" ist entweder falsch geschrieben oder

konnte nicht gefunden werden.
Files\vlang\v\thirdparty\cJSON\cJSON.o not found, building it...
ls() couldnt open dir "Files\vlang\v\thirdparty\cJSON" (does not exist).
Der Befehl "gcc" ist entweder falsch geschrieben oder

konnte nicht gefunden werden.
Der Befehl "gcc" ist entweder falsch geschrieben oder

konnte nicht gefunden werden.
V error: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```

v was successfully build before with MSVC.

Again fooled by lazyness but it doesn't change the main problem (paths are still crippled). And `replace` / with \ seems not to work consequently.

```
Lefass\Documents\tenebris\v\thirdparty\cJSON\cJSON.obj not found, building it (with msvc)...
ls() couldnt open dir "Lefass\Documents\tenebris\v\thirdparty\cJSON" (does not exist).
thirdparty cmd line: ""C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.20.27508\bin\Hostx64\x64\cl.exe" /volatile:ms /Z7 -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt" -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.20.27508\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\shared" /c  /Fo"Lefass\Documents\tenebris\v\thirdparty\cJSON\cJSON.obj" /D_UNICODE /DUNICODE"
Microsoft (R) C/C++-Optimierungscompiler Version 19.20.27508.1 f�r x64
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

cl : Befehlszeile error D8003 : Name der Quelldatei fehlt.
Lefass\Documents\tenebris\v\thirdparty\cJSON\cJSON.obj not found, building it (with msvc)...
ls() couldnt open dir "Lefass\Documents\tenebris\v\thirdparty\cJSON" (does not exist).
thirdparty cmd line: ""C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.20.27508\bin\Hostx64\x64\cl.exe" /volatile:ms /Z7 -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt" -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.20.27508\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\shared" /c  /Fo"Lefass\Documents\tenebris\v\thirdparty\cJSON\cJSON.obj" /D_UNICODE /DUNICODE"
Microsoft (R) C/C++-Optimierungscompiler Version 19.20.27508.1 f�r x64
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

cl : Befehlszeile error D8003 : Name der Quelldatei fehlt.

========== cl cmd line:
""C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.20.27508\bin\Hostx64\x64\cl.exe" -w /we4013 /volatile:ms /D_UNICODE /DUNICODE /Fo"e:\Users\Enrico Lefass\Documents\tenebris\v\examples\news_fetcher.tmp.c.obj" /Z7 /MDd "e:\Users\Enrico Lefass\Documents\tenebris\v\examples\news_fetcher.tmp.c"  -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt"   -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.20.27508\include"   -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um"   -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\shared"   -I "e:\Users\Enrico"   -I "e:\Users\Enrico"  Lefass\Documents\tenebris\v/thirdparty/cJSON e:\Users\Enrico Lefass\Documents\tenebris\v\thirdparty\cJSON\cJSON.obj Lefass\Documents\tenebris\v/thirdparty/vschannel kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib vcruntime.lib ws2_32.lib ws2_32.lib crypt32.lib Urlmon.lib /link /NOLOGO /OUT:"e:\Users\Enrico Lefass\Documents\tenebris\v\examples\news_fetcher.exe" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.17763.0\ucrt\x64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.17763.0\um\x64" /LIBPATH:"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.20.27508\lib\x64" /INCREMENTAL:NO /DEBUG:FULL"
==========

V error: Microsoft (R) C/C++-Optimierungscompiler Version 19.20.27508.1 f�r x64
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

cl : Befehlszeile warning D9024 : Unbekannter Typ der Quelldatei "Lefass\Documents\tenebris\v/thirdparty/cJSON", Objektdatei wird angenommen.
cl : Befehlszeile warning D9024 : Unbekannter Typ der Quelldatei "e:\Users\Enrico", Objektdatei wird angenommen.
cl : Befehlszeile warning D9024 : Unbekannter Typ der Quelldatei "Lefass\Documents\tenebris\v/thirdparty/vschannel", Objektdatei wird angenommen.
news_fetcher.tmp.c
e:\Users\Enrico Lefass\Documents\tenebris\v\examples\news_fetcher.tmp.c(124): fatal error C1083: Datei (Include) kann nicht ge�ffnet werden: "cJSON.h": No such file or directory
```

--- a/compiler/msvc.v
+++ b/compiler/msvc.v
@@ -346,7 +346,7 @@ pub fn (v mut V) cc_msvc() {
 				rest = ''
 				base.trim_space()
 			}
-			println('arg: $arg')
+
 			flags << ParsedFlag {
 				fl, arg
 			}

--- a/compiler/msvc.v
+++ b/compiler/msvc.v
@@ -331,7 +331,7 @@ pub fn (v mut V) cc_msvc() {
 			// TODO: we really shouldnt support all of these cmon
 			mut lowest := base.index('-')
 			// make sure hyphen comes before I|l|L to not break paths with hyphens
-			if lowest < base.len-2 && !(base[lowest+1] in [`I`, `l`, `L`]) {
+			if lowest != -1 && lowest < base.len-2 && !(base[lowest+1] in [`I`, `l`, `L`]) {
 				lowest = -1
 			}
 			for x in [base.index(' '), base.index(',')] {

--- a/compiler/msvc.v
+++ b/compiler/msvc.v
@@ -330,6 +330,10 @@ pub fn (v mut V) cc_msvc() {
 			// Which ever one of these is lowest we use
 			// TODO: we really shouldnt support all of these cmon
 			mut lowest := base.index('-')
+			// make sure hyphen comes before I|l|L incase there is hyphen in a path
+			if lowest < base.len-2 && !(base[lowest+1] in [`I`, `l`, `L`]) {
+				lowest = -1
+			}
 			for x in [base.index(' '), base.index(',')] {
 				if (x < lowest && x != -1) || lowest == -1 {
 					lowest = x

--- a/compiler/msvc.v
+++ b/compiler/msvc.v
@@ -330,7 +330,7 @@ pub fn (v mut V) cc_msvc() {
 			// Which ever one of these is lowest we use
 			// TODO: we really shouldnt support all of these cmon
 			mut lowest := base.index('-')
-			// make sure hyphen comes before I|l|L incase there is hyphen in a path
+			// make sure hyphen comes before I|l|L to not break paths with hyphens
 			if lowest < base.len-2 && !(base[lowest+1] in [`I`, `l`, `L`]) {
 				lowest = -1
 			}

--- a/compiler/msvc.v
+++ b/compiler/msvc.v
@@ -330,8 +330,8 @@ pub fn (v mut V) cc_msvc() {
 			// Which ever one of these is lowest we use
 			// TODO: we really shouldnt support all of these cmon
 			mut lowest := base.index('-')
-			// make sure hyphen comes before I|l|L to not break paths with hyphens
-			if lowest != -1 && lowest < base.len-2 && !(base[lowest+1] in [`I`, `l`, `L`]) {
+			// dont break paths with hyphens
+			if lowest != 0  {
 				lowest = -1
 			}
 			for x in [base.index(' '), base.index(',')] {
@@ -346,7 +346,7 @@ pub fn (v mut V) cc_msvc() {
 				rest = ''
 				base.trim_space()
 			}
-
+			println('arg: $arg')
 			flags << ParsedFlag {
 				fl, arg
 			}

